test/end-to-end/company.js

```diff
@@ -227,6 +228,290 @@ describe("Company endpoint", () => {
+                        test_company_data, disabled_company_data, blocked_company_data]);
+
+                await Account.deleteMany({});
+
+                await Account.create({
+                    email: test_user_admin.email,
+                    password: await hash(test_user_admin.password),
+                    isAdmin: true
+                });
+
+                await Account.create({
+                    email: test_user.email,
+                    password: await hash(test_user.password),
+                    company: test_company_4._id
+                });
+            });
+
+            afterEach(async () => {
+                await test_agent
+                    .delete("/auth/login")
+                    .expect(HTTPStatus.OK);
+            });
+
+            test("should fail if not a valid id", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(HTTPStatus.OK);
+
+                const res = await test_agent
+                    .get("/company/123/profile")
+                    .send({ adminReason })
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+                expect(res.body).toHaveProperty("error_code", ErrorTypes.VALIDATION_ERROR);
+                expect(res.body).toHaveProperty("errors");
+                expect(res.body.errors[0]).toHaveProperty("param", "companyId");
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.OBJECT_ID);
+            });
+
+            test("should fail if company does not exist", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(HTTPStatus.OK);
+
+                const id = "111111111111111111111111";
+                const res = await test_agent
+                    .get(`/company/${id}/profile`)
+                    .send({ adminReason })
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+                expect(res.body).toHaveProperty("error_code", ErrorTypes.VALIDATION_ERROR);
+                expect(res.body).toHaveProperty("errors");
+                expect(res.body.errors[0]).toHaveProperty("param", "companyId");
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.COMPANY_NOT_FOUND(id));
+            });
+
+            test("should succeed if the company has no offers", async () => {
+                const res = await test_agent
+                    .get(`/company/${test_company_2.id}/profile`)
+                    .expect(HTTPStatus.OK);
+                expect(res.body).toHaveProperty("offers");
+                expect(res.body.offers).toHaveProperty("length", 0);
+            });
+
+            test("should return all offers if below the maximum limit", async () => {
```

Comment:
"Should return all company offers when company has less offers than the limit" or something similar.